File: src/main/java/com/tecode/house/zouchao/dao/HBaseDao.java
package com.tecode.house.zouchao.dao;

import scala.Tuple2;
import scala.Tuple3;

import java.io.IOException;
import java.util.List;
import java.util.Map;

public interface HBaseDao {
    /**
     * 获取公平市场租金
     *
     * @param tableName 表名
     * @return
     * @throws IOException
     */
    List<Integer> getAllRent(String tableName) throws IOException;

    List<Tuple2<String, Integer>> getAllRentByCreateYear(String tableName) throws IOException;

    List<Tuple3<String, Integer, Integer>> getAllRoomByCreateYear(String tableName);
}

File: src/main/java/com/tecode/house/lijun/dao/HBaseDao.scala
package com.tecode.house.lijun.dao

import java.util

trait HBaseDao {
  /**
    * 查询租金报表的所有数据
    *
    * @param tableName 表名
    * @param page      查询的页码
    * @return (Int,util.List[util.ArrayList[String])：（符合条件的数据的总条数，查询页码的数据）
    **/
  def getForCOST(tableName: String, page: Int): (Int, util.List[util.ArrayList[String]])


  /**
    * 按建成年份统计价格查询所有数据
    *
    * @param tableName 表名
    * @param page      查询页码
    * @return (Int,util.List[util.ArrayList[String])：（符合条件的数据的总条数，查询页码的数据）
    **/

  def getAllForValue(tableName: String, page: Int): (Int, util.List[util.ArrayList[String]])


  /**
    * 按建成年份统计房屋数量过滤查询数据(只过滤城市规模)
    *
    * @param tableName 表名
    * @param page      查询页码
    */
  def getForIncome(tableName: String, page: Int): (Int, util.List[util.ArrayList[String]])


}
